doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md

```diff
@@ -155,7 +155,71 @@ the boolean argument specifies whether to generate public/private key pair autom
 The first argument can be either `DesktopKaaPlatformContext` for Java desktop or `AndroidKaaPlatformContext` for Android applications. 
 In this example, the second argument is the default implementation of `KaaClientStateListener` -- `SimpleKaaClientStateListener`, that is solely used to log the client state changes.
 And last one if set to **false** -- means using existing pre-generated public/private key pair, **true** -- means auto-generating new public/private key pair if absent.
+
+You can configure properties and `ExecutorContext` of `DesktopKaaPlatformContext` with help of it's constructors. `DesktopKaaPlatformContext` uses `FlexibleExecutorContext` implementation of `ExecutorContext` by default. You can change it by calling corresponding constructor with specific `ExecutorContext` implementation.
+
+`FlexibleExecutorContext` has four thread groups, which responsible for life cycle, callbacks, API and scheduling. Each of these groups is implemented as separate thread pool (using ThreadPoolExecutor or its subclass). Each of them has minimum and maximum threads amount and maximum thread idle time. By default minimum threads amount for all these groups is zero, maximum threads amount is almost not limited (it set to `Integer.MAX_VALUE`) and maximum idle time is 100 milliseconds. You can call corresponding constructor to limit maximum threads amount or idle time (minimum threads amount is always zero). For scheduled thread groups, you can change only it's minimum threads amount. It is recommended to use builder to construct the `FlexibleExecutorContext` instance with custom parameters.
```

Comment:
s/which responsible for life cycle, callbacks, API and scheduling/responsible for life cycle, callbacks, API, and scheduling